infra/azure/terraform/README.md

```diff
@@ -0,0 +1,62 @@
+# infra/azure
+Terraform scripts and documentation for Cluster API for Azure ([CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/templates)) infrastructure that the Kubernetes community runs on Azure.
+The capz folder structure is formatted shown below:
+main.tf
+├── resource_group
+├── identities
+├── key_vault
+├── container_registry
+├── storage_account
+├── role_assignments
+├── variables
+
+# Prerequiste 
+The `az-cli-prow` service principal with federated credentials is required to be created by a tenant admin. The [iam-config\gce-param.json](./iam-config/gce-param.json) and [iam-config\eks-param.json](./iam-config/eks-param.json) can be used for creating the fededrated credentials for GCE and EKS. Below is an example of how to add a federated credential.
+
+    ```sh
+    az ad sp create-for-rbac --name az-cli-prow

```

Comment:
Since you're already creating new user assigned identities anyways for the key-vault and assigning its permissions, it would be nice if you just went ahead and created the identity needed for `az-cli-prow` as a user assigned MSI versus having the existing the service principal reference which you then assign write permissions to the subscription.

Reply:
`az-cli-prow` identity needs to be a SP to be able to log in with the Azure and be used to create resources.  Azure cli can't login in vai Federated credentials and Managed Identities